cmd/operator-sdk/main.go

```diff
@@ -22,21 +22,25 @@ import (
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 
 	"github.com/operator-framework/operator-sdk/cmd/operator-sdk/cli"
-	"github.com/operator-framework/operator-sdk/internal/flags"
+	kbutil "github.com/operator-framework/operator-sdk/internal/util/kubebuilder"
 
 	log "github.com/sirupsen/logrus"
-	"github.com/spf13/viper"
 )
 
 func main() {
-	root := cli.GetCLIRoot()
-
-	root.PersistentFlags().Bool(flags.VerboseOpt, false, "Enable verbose logging")
-	if err := viper.BindPFlags(root.PersistentFlags()); err != nil {
-		log.Fatalf("Failed to bind root flags: %v", err)
+	// Use the new KB CLI only when running inside an existing Kubebuilder project with a PROJECT file.
+	// The default legacy CLI provides the init cmd to initialize
+	// a Kubebuilder project as a way to opt into the new KB CLI.
+	// TODO: Make the new KB CLI the default, once the integration is complete
+	// and deprecate "operator-sdk new" from the old CLI.
+	if kbutil.IsConfigExist() {
```

Comment:
@estroz This is what I meant by reversing the default CLI from what you had in #2779
With this, the legacy CLI is still the default when you're scaffolding a new project.
And the legacy CLI now provides the `operator-sdk init` command as a hidden subcommand after which you switch over to the new CLI.
https://github.com/operator-framework/operator-sdk/pull/2827/files#diff-48b64768aa84f740e1b924dffaf0f205R94

Let me know what you think.

Reply:
Ah I see you did the opposite of #2779. That's fine with me.